src/Native/System.Native/pal_networking.cpp

```diff
@@ -624,3 +638,77 @@ SetIPv6Address(uint8_t* socketAddress, int32_t socketAddressLen, uint8_t* addres
 
     return PAL_SUCCESS;
 }
+
+extern "C" int32_t GetControlMessageBufferSize(int32_t isIPv4, int32_t isIPv6)
+{
+    return (isIPv4 != 0 ? CMSG_SPACE(sizeof(in_pktinfo)) : 0) + (isIPv6 != 0 ? CMSG_SPACE(sizeof(in6_pktinfo)) : 0);
+}
+
+static int32_t GetIPv4PacketInformation(cmsghdr* controlMessage, IPPacketInformation* packetInfo)
+{
+    assert(controlMessage != nullptr);
+    assert(packetInfo != nullptr);
+
+    if (controlMessage->cmsg_len < sizeof(in_pktinfo))
```

Comment:
Should this be an assert? If this is called with a different type, that should be an upstream issue right? Same with the IPv6 one below